src/overlays/actors/ovl_En_Dt/z_en_dt.c

```diff
@@ -28,59 +95,714 @@ ActorInit En_Dt_InitVars = {
+        this->textIdIndex = 23; // "You take refuge too"
+        this->visualState = 5;
+        this->meetingFinished = true;
+        EnDt_UpdateVisualState(this);
+        Message_BombersNotebookQueueEvent(play, BOMBERS_NOTEBOOK_PERSON_MAYOR_DOTOUR);
+    }
+
+    if (this->visualState == 0) {
+        if (CHECK_WEEKEVENTREG(WEEKEVENTREG_60_08)) {
+            this->textIdIndex = 9; // "All must take refuge!!!"
+            this->visualState = 2;
+        }
+
+        EnDt_UpdateVisualState(this);
+        if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_60_08)) {
+            this->visualState = 1;
+        }
+    }
+
+    this->actor.textId = sTextIds[this->textIdIndex];
+    if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_63_80)) {
+        EnDt_UpdateCutsceneFocusTarget(this);
+    }
+
+    actionFn = EnDt_OfferRegularTalk;
+    this->npcState = EN_DT_NPC_STATE_IDLE;
+    this->actionFunc = actionFn;
+}
+
+void EnDt_OfferRegularTalk(EnDt* this, PlayState* play) {
+    EnMuto* npcMuto = NULL;
+    EnBaisen* npcBaisen = NULL;
+
+    if (Actor_TalkOfferAccepted(&this->actor, &play->state)) {
+        EnDt_SetupMeetingCutscene(this, play);
+        return;
+    }
+
+    if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_63_80)) {
+        if (this->npcEnMuto != NULL && this->npcEnBaisen) {
+            npcMuto = (EnMuto*)this->npcEnMuto;
+            npcBaisen = (EnBaisen*)this->npcEnBaisen;
+        }
+
+        this->showedCouplesMask = false;
+        this->textIdIndex = 0;
+
+        if (Player_GetMask(play) == PLAYER_MASK_COUPLE) {
+            this->textIdIndex = 11; // "Ah!"
+            this->showedCouplesMask = true;
+
+            if (this->npcEnMuto && this->npcEnBaisen) {
+                npcMuto->textIdIndex = 4;   // "Ah!"
+                npcBaisen->textIdIndex = 6; // "Ah!"
+            }
+        } else if (CHECK_WEEKEVENTREG(WEEKEVENTREG_60_08)) {
+            this->textIdIndex = 9; // "All must take refuge!!!"
+        }
+    }
+
+    this->actor.textId = sTextIds[this->textIdIndex];
+    if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_63_80) && this->npcEnMuto != NULL && this->npcEnBaisen != NULL &&
+        (npcMuto->cutsceneState == 1 || npcBaisen->cutsceneState == 1)) {
+        EnDt_SetupMeetingCutscene(this, play);
+    }
+
+    // After completing Couple's Mask event and wearing Kafeis Mask
+    if (CHECK_WEEKEVENTREG(WEEKEVENTREG_63_80) && (Player_GetMask(play) == PLAYER_MASK_KAFEIS_MASK)) {
+        this->actor.textId = 0x2368; // "My wife hired you? Go ask the Curiosity Shop"
+    }
+
+    Actor_OfferTalk(&this->actor, play, 150.0f);
+}
+
+void EnDt_SetupMeetingCutscene(EnDt* this, PlayState* play) {
+    EnMuto* npcMuto;
+    EnBaisen* npcBaisen;
+    s32 index;
+    s16* pTextData;
+
+    if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_63_80)) {
+        if (this->npcEnMuto != NULL) {
+            if (this->npcEnBaisen != NULL) {
+                npcMuto = (EnMuto*)this->npcEnMuto;
+                npcBaisen = (EnBaisen*)this->npcEnBaisen;
+            }
+        }
```

Comment:
```suggestion
        if ((this->npcEnMuto != NULL) && (this->npcEnBaisen != NULL)){
            npcMuto = (EnMuto*)this->npcEnMuto;
            npcBaisen = (EnBaisen*)this->npcEnBaisen;
        }
```
Does it still match if you merge the two conditions?